terrawrap/utils/config.py

```diff
@@ -86,6 +89,152 @@ def parse_wrapper_configs(wrapper_config_files: List[str]) -> WrapperConfig:
+                    if not wrapper_config_obj.config:
+                        continue
+                    if wrapper_config_obj.depends_on is None:
+                        post_graph_runs.append(dir_path)
+                        continue
+                    single_config_dependency_graph = networkx.DiGraph()
+                    visited: List[str] = []
+                    graph_wrapper_dependencies(dir_path, config_dict, single_config_dependency_graph, visited)
+                    graph_list.append(single_config_dependency_graph)
+            if not has_tf_wrapper and is_config_directory(dir_path):
+                post_graph_runs.append(dir_path)
+
+    directory_graph = networkx.compose_all(graph_list)
+    return directory_graph, post_graph_runs
+
+
+# pylint: disable=R0912
+def graph_wrapper_dependencies(config_dir: str, config_dict, graph: networkx.DiGraph, visited: List[str]):
+    """
+    Given a directory, recursively finds all other directories it depends on and builds a graph.
+    :param config_dir: The config directory to obtain a dependency graph for
+    :param config_dict: A dictionary containing wrapper config objects for each seen directory
+    :param graph: The graph to add dependency info to. Empty at first, reused in recursion.
+    :param visited: A list of visited nodes. Empty at first, reused in recursion
+    """
+    if config_dir in visited:
+        return
+    visited.append(config_dir)
+
+    if config_dict.get(config_dir):  # add to dictionary so we only read the file once
+        wrapper_config_obj = config_dict[config_dir].get("wrapper_config")
+    else:
+        wrapper_config_obj = create_wrapper_config_obj(config_dir)
+        config_dict[config_dir] = {
+            "wrapper_config": wrapper_config_obj
+        }
+
+    if wrapper_config_obj.config:
+        graph.add_node(config_dir)
+
+    tf_dependencies = wrapper_config_obj.depends_on
+
+    if tf_dependencies is None:
+        print("Cannot list a dependency without tf_wrapper dependency configuration:", config_dir)
+        sys.exit(1)
+
+    for dependency in tf_dependencies:
+        graph.add_node(dependency)
+        if config_dir in graph:
+            graph.add_edge(dependency, config_dir)
+
+    wrappers = find_wrapper_config_files(config_dir)
+    wrappers.reverse()  # we want the closest wrapper file that gives inherited dependencies
+    for wrapper in wrappers:
+        wrapper_dir = os.path.dirname(wrapper)
+        if config_dict.get(wrapper_dir):  # add to dictionary so we only read the file once
+            new_wrapper_config_obj = config_dict[wrapper_dir].get("wrapper_config")
+        else:
+            new_wrapper_config_obj = create_wrapper_config_obj(wrapper_dir)
+            config_dict[wrapper_dir] = {
+                "wrapper_config": new_wrapper_config_obj
+            }
+
+        if wrapper_dir == config_dir:
+            continue
+        if new_wrapper_config_obj.depends_on is not None:
```

Comment:
new_wrapper_config_obj is the closest wrapper file above the current directory in the recursion. This lets us inherit dependencies. This lets us put dependency info for multiple things without having to copy paste. lets say we are in `foo/bar` and foo contains bar/, bar2/ and bar3/. If all of those depend on IAM we can list that in foo/.tf_wrapper and it will be given to all 3 instead of writing it manually in all 3. This works regardless of if foo is a config dir or not.